pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -118,8 +119,13 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 				copied.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
 				copied.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledNewerDeploymentExists
 
-				updatedDeployment, err := c.rn.ReplicationControllers(copied.Namespace).Update(copied)
-				if err != nil {
+				// Retry faster on conflicts
```

Comment:
This part of code is (should be) reentrant. If we end up here, there is a newer version for a config (and version is not decrementing) and we need to cancel this RC no matter what.